### ISSUE TITLE
Update framework conditional compilation directives

### ISSUE THREAD

**Issue by halter73** [aspnet/KestrelHttpServer#2624]
We want compilation to fail if we start targeting a framework not accounted for in the conditional compilation directives. [This](https://github.com/aspnet/MusicStore/blob/d971bfc8b777ca4098a951727218aef32cb9ebd5/test/MusicStore.E2ETests/Common/Helpers.cs#L51-L57) is an example of how to do this.

See https://github.com/aspnet/KestrelHttpServer/pull/2622/files#r191625156. We can make this change as soon as #2622 is merged into the dev branch.

**Comment by halter73** [comment]
I just merged #2622 into dev (5 minutes ago it was only in the release/2.1 branch) so you should really be unblocked @Tratcher.
